docs/standard/native-interop/tutorial-comwrappers.md

```diff
@@ -0,0 +1,430 @@
+
+```csharp
+var cw = new DemoComWrappers();
+var demo = new DemoImpl();
+IntPtr ccw = cw.GetOrCreateComInterfaceForObject(demo, CreateComInterfaceFlags.None);
+```
+
+### Step 3 &ndash; Implement `CreateObject()`
+
+Constructing a Native Object Wrapper has more implementation options and a great deal more nuance than constructing a Managed Object Wrapper. The first question to address is how permissive the `ComWrappers` subclass will be in supporting COM types. To support all COM types, which is possible, you'll need to write a substantial amount of code or employ some clever uses of `Reflection.Emit`. For this tutorial, you'll only support COM instances that implement both `IDemoGetType` and `IDemoStoreType`. Since you know there is a finite set and have restricted that any supplied COM instance must implement both interfaces, you could provide a single, statically defined wrapper; however, dynamic cases are common enough in COM that we'll explore both options.
+
+#### Static Native Object Wrapper
+
+Let's look at the static implementation first. The static Native Object Wrapper involves defining a managed type that implements the .NET interfaces and can forward the calls on the managed type to the COM instance. A rough outline of the static wrapper follows.
+
+```csharp
+class DemoNativeStaticWrapper
+    : IDemoGetType
+    , IDemoStoreType
+{
+    public string? GetString() =>
+        throw new NotImplementedException();
+
+    public void StoreString(int len, string? str) =>
+        throw new NotImplementedException();
+}
+```
+
+To construct an instance of this class and provide it as a wrapper, you must define some policy. If this type is used as a wrapper, it would seem that since it implements both interfaces, the underlying COM instance should implement both interfaces too. Given that you're adopting this policy, you'll need to confirm this through calls to [`Marshal.QueryInterface()`][api_marshalqueryinterface] on the COM instance.
+
+```csharp
+int hr = Marshal.QueryInterface(ptr, ref IDemoGetType.IID_IDemoGetType, out IntPtr IDemoGetTypeInst);
+if (hr != 0)
+{
+    return null;
+}
+
+hr = Marshal.QueryInterface(ptr, ref IDemoStoreType.IID_IDemoStoreType, out IntPtr IDemoStoreTypeInst);
+if (hr != 0)
+{
+    Marshal.Release(ptr);
+    return null;
+}
+
+return new DemoNativeStaticWrapper()
+{
+    IDemoGetTypeInst = IDemoGetTypeInst,
+    IDemoStoreTypeInst = IDemoStoreTypeInst
+};
+```
+
+#### Dynamic Native Object Wrapper
+
+Dynamic wrappers are more flexible because they provide a way for types to be queried at run time instead of statically. In order to provide this support, you'll utilize [`IDynamicInterfaceCastable`][api_idynamicinterfacecastable]. Observe that `DemoNativeDynamicWrapper` only implements this interface. The functionality that the interface provides is a chance to determine what type is supported at run time.
```

Comment:
> types to be queried at run time instead of statically.

I think this is confusing. Even the static one queries the types at runtime. If I understand this correctly the core difference is that with the static one all of the COM interfaces and their managed counterparts must be figured out up front (for each instance it's needed on), the dynamic one is lazy, it only needs to figure out the managed interface when something actually asks for it.

Curiosity: How does the `IDynamicallyInterfaceCastable` play with reflection - for example what happens if I call `obj.GetType().GetInterfaces()` - will it include the dynamic interfaces or not?